datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -231,19 +244,43 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
   }
 
   internal fun goToPreviousPage() {
-    val previousPageIndex =
-      pages!!.indexOfLast { it.index < currentPageIndexFlow.value!! && it.enabled }
-    check(previousPageIndex != -1) {
-      "Can't call goToPreviousPage() if no preceding page is enabled"
+    when (entryMode) {
+      EntryMode.PRIOR_EDIT, EntryMode.RANDOM -> {
+        val previousPageIndex =
+          pages!!.indexOfLast { it.index < currentPageIndexFlow.value!! && it.enabled }
+        check(previousPageIndex != -1) {
+          "Can't call goToPreviousPage() if no preceding page is enabled"
+        }
+        currentPageIndexFlow.value = previousPageIndex
+      }
+      else -> {
+        Timber.w("Previous questions and submitted answers cannot be viewed or edited.")
+      }
     }
-    currentPageIndexFlow.value = previousPageIndex
   }
 
   internal fun goToNextPage() {
-    val nextPageIndex =
-      pages!!.indexOfFirst { it.index > currentPageIndexFlow.value!! && it.enabled }
-    check(nextPageIndex != -1) { "Can't call goToNextPage() if no following page is enabled" }
-    currentPageIndexFlow.value = nextPageIndex
+    when (entryMode) {
+      EntryMode.PRIOR_EDIT, EntryMode.SEQUENTIAL -> {
+        // Force update validation results for all questions on the current page
+        isPaginationButtonPressed = true
+        modificationCount.update { it + 1 }
```

Comment:
do you think this makes sense? i think we can save some additional validation if the user has already pressed the button.

```suggestion
        if (!isPaginationButtonPressed) {
          // Force update validation results for all questions on the current page
          isPaginationButtonPressed = true
          modificationCount.update { it + 1 }
        }
```

Reply:
If i could map a case 
1. User opened the questionnaire and taps on the next button without touching any items
2. isPaginated is FALSE, it will trigger the validation for the current page items 
3. now errors are shown the required fields, now user tried to tap again on the next button 
4. since isPaginated is already TRUE, this will save the additional validation in this _case_. 
You are right @jingtang10 . This makes sense to me.

cc @maimoonak